features/FEATURE_BLE/source/generic/GenericGap.cpp

```diff
@@ -1142,112 +1175,126 @@ void GenericGap::on_advertising_report(const pal::GapAdvertisingReportEvent& e)
         pal::GapAdvertisingReportEvent::advertising_t advertising = e[i];
 
         // Check if the address hasn't been resolved
-        if(_privacy_enabled && _central_privacy_configuration.resolution_strategy == CentralPrivacyConfiguration_t::RESOLVE_AND_FILTER)
-        {
-            if(advertising.address_type == pal::connection_peer_address_type_t::RANDOM_ADDRESS)
-            {
+        if(_privacy_enabled &&
+            _central_privacy_configuration.resolution_strategy == CentralPrivacyConfiguration_t::RESOLVE_AND_FILTER &&
+            advertising.address_type == pal::connection_peer_address_type_t::RANDOM_ADDRESS &&
+            is_random_private_resolvable_address(advertising.address.data())
+        ) {
                 // Filter it out
-                continue;
-            }
+            return;
```

Comment:
What happens if we have multiple advertising reports?

Reply:
Good catch; continue is more appropriate here.

Reply:
fixed in my repo